src/lsm/table.zig

```diff
@@ -432,18 +398,56 @@ pub fn TableType(
                     .op = options.address,
                     .timestamp = options.snapshot_min,
                     .request = builder.value_count,
-                    .size = block_size - @as(u32, @intCast(values_padding.len + block_padding.len)),
+                    .size = @sizeOf(vsr.Header) + builder.value_count * @sizeOf(Value),
                     .command = .block,
                     .operation = schema.BlockType.data.operation(),
                 };
 
                 header.set_checksum_body(block[@sizeOf(vsr.Header)..header.size]);
                 header.set_checksum();
 
+                const values = Table.data_block_values_used(block);
+                { // Now that we have checksummed the block, sanity-check the result:
+
+                    if (constants.verify) {
+                        var a = &values[0];
+                        for (values[1..]) |*b| {
+                            assert(compare_keys(key_from_value(a), key_from_value(b)) == .lt);
+                            a = b;
+                        }
+                    }
+
+                    assert(builder.value_count == values.len);
+                    assert(block_size - header.size ==
+                        (data.value_count_max - values.len) * @sizeOf(Value) + data.padding_size);
+                    // Padding is short on average, so assert unconditionally.
+                    assert(stdx.zeroed(block[header.size..]));
```

Comment:
That's the main value in this PR. As far as I can tell, we don't have an equivalent assert anywhere. 

It wasn't exactly clear what's the best place to add this assert in this method initially, so I reshufled code a bit to make it clear when we change what.

Reply:
:thinking: I wonder if we do actually zero the padding here? I haven't found that code yet!

Reply:
Hm, yeah, I think this is a buffer bleed! In a table with two data blocks, if the second block is only partially filled, the suffix will include the data from the previous data block!

Reply:
There may have been an implicit zeroing by Grid's `allocate_block()`, but within compaction, and when cycling blocks, that indeed sounds like a bleed. Nice find!

Reply:
Since this is data plane, we should gate the assertion on `verify`, as part of online verification.

Reply:
See the comment above this line! I originally gated this with verify, but then reasoned that this is padding, so it actually is fine to check unconditionally: 

* if the block is full, there's not much to check
* if the block is empty, we'd have already saved some time due to not filling the block, so we can use that credit to check the padding!

Reply:
Ah, I had missed that comment. This is much better as you have it now.

Reply:
For posterity: there's no buffer bleed here, here's where we do the zeroing,

https://github.com/tigerbeetle/tigerbeetle/blob/8657375bb8f18bc022f0a5c6b5b04f8db6ad7b26/src/vsr/grid.zig#L604-L605

, but it takes a [two star programmer](https://wiki.c2.com/?ThreeStarProgrammer) to see that.